yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+    fatal error: Python.h: No such file or directory
+| 28 | #include <Python.h>
+| | ^~~~~~~~~~
+| compilation terminated.
+| ninja: build stopped: subcommand failed.
+```]
+
+---
+# Build troubleshooting - custom applications
+
+* Custom applications often are provided by our Clients
+  - they were compiled in some environment
+  - we integrate them for given Yocto release
+  - update may introduce problems (compilator flags, deprecated variables)
+  - example: using app after Qt5 update to Qt6
+
+.code-9px[```bash
+| /build/tmp-glibc/(...)/git/libs/qmqtt_client/ecc/ecc_manager.cpp: At global scope:
+| /build/tmp-glibc/(...)/git/libs/qmqtt_client/ecc/ecc_manager.cpp:42:13: warning: 'bool
+    writePrivateKeyToFile(const QString&, EVP_PKEY*)' defined but not used [-Wunused-function]
+|    42 | static bool writePrivateKeyToFile(const QString& fileName, EVP_PKEY* key)
+|       |             ^~~~~~~~~~~~~~~~~~~~~
+| make[2]: *** [Makefile:318: temp/ecc_manager.o] Error 1
+| make[2]: Leaving directory '/build/tmp-glibc/(...)/build/libs/qmqtt_client'
+| make[1]: *** [Makefile:47: sub-qmqtt_client-make_first-ordered] Error 2
+| make[1]: Leaving directory '/build/tmp-glibc/(...)/build/libs'
+| make: *** [Makefile:48: sub-libs-make_first-ordered] Error 2
+| ERROR: oe_runmake failed
+| WARNING: exit code 1 from a shell command.
+ERROR: Task (/work/meta-abcd-app/recipes-qt/abcd-mqtt-proxy/abcd-mqtt-proxy_git.bb:do_compile)
+    failed with exit code '1'
+NOTE: Tasks Summary: Attempted 4630 tasks of which 4609 didn't need to be rerun and 1 failed.
+
+Summary: 1 task failed:
+  /work/meta-abcd-app/recipes-qt/abcd-mqtt-proxy/abcd-mqtt-proxy_git.bb:do_compile
+```]
+
+---
+# Veryfing update
+
+* Solving any build problems usually leads us to a bootable image
+  - it is insufficient, many functionalities may not work as they should
+* Documentation and testing are key here
+  - docs allow us to understand expecting behavior
+  - test should cover every funtionality and verify them
+  - they also should fullfill agreeded requirements
+* This is where the most tedious work of updating begins
+  - iterations of tests, patches, rebuilds and retests
+  - CI/CD pipeline helps a lot
+* Tests suites needs also updates
+  - the test result based on an erroneous assumption, which was changed with the
+    update
+  - logs changed behavior
+
+---
+# Veryfing update
+
+* Examples of changes needed in tests, when we got different logs
+
+.code-14px[```bash
+-   path: /usr/bin/tensorflow-lite-2.4.0/examples
++   path: /usr/bin/tensorflow-lite-2.9.1/examples
+```]
+
+.code-14px[```bash
+-   - "PCI bridge: Synopsys, Inc. DWC_usb3 (rev 01)"
++   - "PCI bridge: Synopsys, Inc. DWC_usb3 / PCIe bridge (rev 01)"
+```]
+
+* Change the way how output is achieved
+  - logging level may changed so, we need to run `dmesg`
+  - other commands may not longer print anything to userspace
+
+---
+# Summary
+
+* Performing Yocto metadata updates is a complex and multi-level problem
```

Comment:
Try to shorten text in the slide, please

Reply:
done https://github.com/3mdeb/conferences/pull/7/commits/ee8eb3bdd8a1804b329190b6310ff5dcf615c84b